.github/workflows/deploy-github-pages.yml

```
@@ -0,0 +1,31 @@
+name: Deploy Github Pages
+
+on:
+  push:
+    branches: [ maknop/github-pages-deployment ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [18.x] # We will deploy with only one version of node  
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: npm ci, build and test
+        run: |
+          npm ci
+          npm run build --if-present     
```

You can update the homepage on the fly with this:

```suggestion
          npm ci
          npx json -I -f package.json -e "this.homepage='https://redhatinsights.github.io/api-documentation-frontend/'"
          npm run build --if-present     
```